src/tagged-type.ts

```diff
@@ -0,0 +1,118 @@
+export type Module = string;
+export type DocumentReference = string;
+
+/**
+ * TaggedType provides the encoding/decoding of the Fauna Tagged Type formatting
+ */
+export class TaggedTypeFormat {
+  /**
+   * Encode the Object to the Tagged Type format for Fauna
+   *
+   * @param obj - Object that will be encoded
+   * @returns Map of result
+   */
+  static encode(obj: any): any {
+    return new TaggedTypeEncoded(obj).result;
+  }
+
+  /**
+   * Decode the JSON string result from Fauna to remove Tagged Type formatting.
+   *
+   * @param input - JSON string result from Fauna
+   * @returns object of result of FQL query
+   */
+  static decode(input: string): any {
+    return JSON.parse(input, (_, value: any) => {
+      if (value["@mod"]) {
+        return value["@mod"] as Module;
+      } else if (value["@doc"]) {
+        return value["@doc"] as DocumentReference;
+      } else if (value["@int"]) {
+        return Number(value["@int"]);
+      } else if (value["@long"]) {
+        return Number(value["@long"]);
```

Comment:
I don't think this is safe. JS doesn't support 64 bit integers because all numbers are 64 bit floats in JS. We prob need to branch further if we're not using fauna-vended types, but i'm starting to lean toward fauna-vended types as a way to guarantee consistency and mitigate data loss/confusion.

We talk about this in our docs in case that helps. https://docs.fauna.com/fauna/current/build/fql/types?lang=javascript#large-integers

Reply:
yeah I think we need to vend types, I should have put a TODO comment here as I was unsure of what to do but wanted to keep rolling.

Reply:
modern javascript has a BigInteger type.

IMO we should use that.

Reply:
Added handling of Fauna Longs as BigInt